devmand/gateway/src/devmand/channels/cli/CliFlavour.cpp

```diff
@@ -44,19 +44,18 @@ SemiFuture<Unit> UbiquitiInitializer::initialize(
 Future<string> DefaultPromptResolver::resolvePrompt(
     shared_ptr<SessionAsync> session,
     const string& newline) {
-  return session
-      ->read(-1)
-      .thenValue([=](...) { return resolvePrompt(session, newline, 1); });
+  return session->read().thenValue(
+      [=](...) { return resolvePrompt(session, newline, delayDelta); });
 }
 
 Future<string> DefaultPromptResolver::resolvePrompt(
     shared_ptr<SessionAsync> session,
     const string& newline,
-    int delayCounter) {
-  return resolvePromptAsync(session, newline, delayCounter)
+    chrono::milliseconds delay) {
+  return resolvePromptAsync(session, newline, delay)
       .thenValue([=](Optional<string> prompt) {
         if (!prompt.hasValue()) {
-          return resolvePrompt(session, newline, delayCounter + 1);
+          return resolvePrompt(session, newline, delay + delayDelta);
```

Comment:
Seems the duration of the delay in iterations has changed.

Before: iteration * baseline //where iteration is 1,2,3.. and baseline was 1000ms
Now: baseline + delayDelta

Not saying it's wrong, just noting it has diverged from the original implementation

Reply:
Ack. My view here is that since this is non-blocking code, the cost of 100ms increments (e.g. 100,200,300) is sensible and does not incur cost associated with this call previously.